scripts/paternity-test.py

```diff
@@ -0,0 +1,173 @@
+import io
+import os
+import json # might not be needed
+from firecloud import fiss
+from firecloud.errors import FireCloudServerError
+import firecloud.api as fapi
+import numpy as np
+import pandas as pd
+import logging
+#import re <-- doesn't work on Terra
+
+# User-set variables
+BILLING_PROJECT_ID="biodata-catalyst"
+WORKSPACE="TSV-AFY"
+SUBDIRECTORY="\/case2\/"
+TABLE_NAME="CRAMs_and_CRAIs" #Do not include spaces or weird characters
+PARENT_FILETYPE="cram"
+# If your filenames are like this, please set INCLUDE_PARENT_EXTENSION to True:
+# NWD119844.CRAM
+# NWD119844.CRAM.CRAI
+# If you filenames are like this, please set INCLUDE_PARENT_EXTENSION to False:
+# NWD119844.CRAM
+# NWD119844.CRAI
+# No quotation marks for this variable, just True or False
+INCLUDE_PARENT_EXTENSION = True
+# Make sure to escape the slashes in your SUBDIRECTORY variable. 
+# For instance, if your files are in a folder called "testfiles" you will need to
+# enter it as "/\testfiles/\" or else Python will get angry.
+
+# Call FireCloud API
+try:
+    bucket = os.environ["WORKSPACE_BUCKET"]
+    response = fapi.list_entity_types(BILLING_PROJECT_ID, WORKSPACE)
+    if response.status_code != 200:
+        print("Error in Firecloud, check your billing project ID and the name of your workspace.")
+    else:
+        print("Firecloud has found your workspace!")
+        directory = bucket + SUBDIRECTORY
+except NameError:
+    print("Caught a NameError exception. This probably means you didn't restart the kernal after"
+          " running the first block of code (the one with all the imports). Run it again, restart"
+          " the kernal, then try running every block of code (including the import one) again.")
+
+# Do magic to create a TSV file
+# We don't have the option to use Python's parsing directly, as the notebook VM is seperate 
+# from the Google Cloud bucket where the data is actually stored, and we cannot run arbitrary 
+# code in a Google Cloud bucket. This forces us to parse the output of gsutil ls. 
+# Parsing ls is always a little dodgy, but we really don't have any other option. 
+# As a result, if your filenames contain non-ascii (ie, stuff besides A-Z, a-z, underscores, and dashes)
+# or bizarre characters (ie, newlines) there is a chance this will not work as expected.
+
+# Pandas dataframe method
+# Note: I am sure there is a more efficient way of doing this!
+
+def baseID(filename_string, child_extension):
+    global PARENT_FILETYPE
+    global INCLUDE_PARENT_EXTENSION
```

Comment:
These globals are not necessary to declare.  They're already declared above the function so they're within the scope of what the function can see.